gluten-core/src/main/scala/org/apache/gluten/extension/StrategyOverrides.scala

```diff
@@ -1,213 +0,0 @@
-      if (buildSide.isDefined) {
-        return Seq(
-          joins.BroadcastHashJoinExec(
-            leftKeys,
-            rightKeys,
-            joinType,
-            buildSide.get,
-            condition,
-            planLater(left),
-            planLater(right)))
-      }
-
-      if (
-        forceShuffledHashJoin &&
-        !BackendsApiManager.getSparkPlanExecApiInstance.joinFallback(
-          joinType,
-          left.outputSet,
-          right.outputSet,
-          condition) &&
-        !left.getTagValue(TAG).isDefined &&
-        !right.getTagValue(TAG).isDefined
-      ) {
-        // Force use of ShuffledHashJoin in preference to SortMergeJoin. With no respect to
-        // conf setting "spark.sql.join.preferSortMergeJoin".
-        val (leftBuildable, rightBuildable) =
-          if (BackendsApiManager.getSettings.utilizeShuffledHashJoinHint()) {
-            // Currently, ClickHouse backend can not support AQE, so it needs to use join hint
-            // to decide the build side, after supporting AQE, will remove this.
-            val leftHintEnabled = hintToShuffleHashJoinLeft(hint)
-            val rightHintEnabled = hintToShuffleHashJoinRight(hint)
-            val leftHintMergeEnabled = hint.leftHint.exists(_.strategy.contains(SHUFFLE_MERGE))
-            val rightHintMergeEnabled = hint.rightHint.exists(_.strategy.contains(SHUFFLE_MERGE))
-            if (leftHintEnabled || rightHintEnabled) {
-              (leftHintEnabled, rightHintEnabled)
-            } else if (leftHintMergeEnabled || rightHintMergeEnabled) {
-              // hack: when set SHUFFLE_MERGE hint, it means that
-              // it don't use this side as the build side
-              (!leftHintMergeEnabled, !rightHintMergeEnabled)
-            } else {
-              (
-                BackendsApiManager.getSettings.supportHashBuildJoinTypeOnLeft(joinType),
-                BackendsApiManager.getSettings.supportHashBuildJoinTypeOnRight(joinType))
-            }
-          } else {
-            (canBuildShuffledHashJoinLeft(joinType), canBuildShuffledHashJoinRight(joinType))
-          }
-
-        if (!leftBuildable && !rightBuildable) {
-          return Nil
-        }
-        val buildSide = if (!leftBuildable) {
-          BuildRight
-        } else if (!rightBuildable) {
-          BuildLeft
-        } else {
-          getSmallerSide(left, right)
-        }
-
-        return Option(buildSide)
-          .map {
-            buildSide =>
-              Seq(
-                joins.ShuffledHashJoinExec(
-                  leftKeys,
-                  rightKeys,
-                  joinType,
-                  buildSide,
-                  condition,
-                  planLater(left),
-                  planLater(right)))
-          }
-          .getOrElse(Nil)
-      }
-      Nil
-    }
-  }
-
-  def existsMultiJoins(plan: LogicalPlan, count: Int = 0): Boolean = {
```

Comment:
The code in strategy have a feature to prevent converting multi-joins to shj. Is it still exited in the new code ?

Reply:
My mistake, I will add it to the column rule later.

Reply:
One more question, why did fallback multi joins need to be used before?

Reply:
Q72 has several joins in the same stage, it shows very bad performance if we use hash join. Fallback multi joins and fallback multi codegens is a temp solution to fallback to Vanilla Spark on such stages. #682 

Now that joinSelectionOverride has been removed, fallback multi joins seem no longer needed, FallbackMultiCodegens is enough. cc @zhztheplayer

Reply:
If I remember correctly that two switches are cooperated for falling back to the original vanilla plan. If we could get that plan in columnar rule after this change, then I think we will no longer need that code.

cc @PHILO-HE

Reply:
cc @JkSelf @FelixYBW

Reply:
tpcds Q72 performance is normal.